src/neo/IO/Caching/DataCache.cs

```diff
@@ -111,12 +114,19 @@ public void Delete(TKey key)
         {
             lock (dictionary)
             {
+                changeSet.Add(key);
+
                 if (dictionary.TryGetValue(key, out Trackable trackable))
                 {
                     if (trackable.State == TrackState.Added)
+                    {
                         dictionary.Remove(key);
+                        changeSet.Remove(key);
```

Comment:
`And` and `Remove`? We can optimize this.

Reply:
Done.